Prefabs/Interactions/Controllables/SharedResources/Scripts/RotationalJointDrive.cs

```diff
@@ -0,0 +1,224 @@
+            float actualTargetAngle = Mathf.Lerp(DriveLimits.minimum, DriveLimits.maximum, facade.TargetValue);
+
+            if (circleUpperLeftQuadrant.Contains(previousActualAngle) && circleUpperRightQuadrant.Contains(currentActualAngle))
+            {
+                rotationMultiplier++;
+            }
+            else if (circleUpperRightQuadrant.Contains(previousActualAngle) && circleUpperLeftQuadrant.Contains(currentActualAngle))
+            {
+                rotationMultiplier--;
+            }
+
+            if (!joint.useLimits && ApplyLimits())
+            {
+                jointRigidbody.velocity = Vector3.zero;
+                jointRigidbody.angularVelocity = Vector3.zero;
+            }
+
+            pseudoRotation = currentActualAngle + (circleDegrees * rotationMultiplier);
+            jointMotor.targetVelocity = (facade.MoveToTargetValue && !pseudoRotation.ApproxEquals(actualTargetAngle, targetValueReachedThreshold) ? facade.DriveSpeed : 0f) * CalculateDirectionMultiplier();
+            joint.motor = jointMotor;
+
+            if (facade.MoveToTargetValue && jointMotor.targetVelocity == 0f)
+            {
+                joint.transform.localRotation = Quaternion.Euler(joint.axis * actualTargetAngle);
+            }
+        }
+
+        /// <inheritdoc />
+        public override Vector3 CalculateDriveAxis(DriveAxis driveAxis)
+        {
+            if (!isActiveAndEnabled)
+            {
+                return Vector3.zero;
+            }
+
+            Vector3 axisDirection = base.CalculateDriveAxis(driveAxis);
+            joint.axis = axisDirection * -1f;
+            return axisDirection;
+        }
+
+        /// <inheritdoc />
+        public override FloatRange CalculateDriveLimits(FloatRange newLimit)
+        {
+            return newLimit;
+        }
+
+        /// <inheritdoc />
+        public override void CalculateHingeLocation(Vector3 newHingeLocation)
+        {
+            if (!isActiveAndEnabled)
+            {
+                return;
+            }
+
+            joint.anchor = newHingeLocation;
+            SetJointContainerPosition();
+        }
+
+        /// <inheritdoc />
+        public override void ProcessDriveSpeed(float driveSpeed, bool moveToTargetValue)
+        {
+        }
+
+        /// <inheritdoc />
+        public override void SetTargetValue(float normalizedValue)
+        {
+        }
+
+        /// <inheritdoc />
+        public override void ProcessAutoDrive(bool autoDrive)
+        {
+            if (!isActiveAndEnabled)
+            {
+                return;
+            }
+
+            joint.useMotor = autoDrive;
+        }
+
+        /// <inheritdoc />
+        protected override float CalculateValue(DriveAxis givenAxis, FloatRange givenLimits)
+        {
+            if (!isActiveAndEnabled)
+            {
+                return 0f;
+            }
+
+            return Mathf.Clamp(pseudoRotation, givenLimits.minimum, givenLimits.maximum);
```

Comment:
This looks like a candidate to be moved into the superclass `RotationalDrive` instead, but I think we can defer this until we do another subclass of a `RotationalDrive` (i.e. the artificial one). That's when we should refactor into the superclass as soon as we notice shared patterns among both subclasses.